### PR TITLE
atypical-padding: FIPS compatibility

### PR DESCRIPTION
**Pull request by tomato42**
### Description

make the test-atypical-padding.py interoperable with servers working in FIPS mode

### Motivation and Context

work towards https://github.com/tlsfuzzer/tlsfuzzer/issues/563

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [ ] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see CI results)
- [ ] [test script checklist](https://github.com/tlsfuzzer/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [ ] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL
  - [ ] NSS
  - [ ] GnuTLS
- [ ] required version of tlslite-ng updated in requirements.txt and README.md

- - -
This change is [Reviewable](https://reviewable.io/reviews/tlsfuzzer/tlsfuzzer/935)

### CODE REVIEW

File: scripts/test-atypical-padding.py [hunk 1]
```diff
@@ -422,6 +451,8 @@ def main():
     node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
     extensions = {ExtensionType.encrypt_then_mac:None,
```
Comment (ep69) [inline scripts/test-atypical-padding.py:452]: This part is a bit confusing when there is `ext` couple of lines above and `extensions` here. What about using something like `ext_ch` and `ext_sh` for example? (Relevant in more places.)

File: scripts/test-atypical-padding.py [hunk 2]
```diff
@@ -219,6 +233,8 @@ def main():
...
                   ExtensionType.renegotiation_info:None}
+    if ems:
+        extensions[ExtensionType.extended_master_secret] = None
     node = node.add_child(ExpectServerHello(extensions=extensions))
```
Comment (ep69) [inline scripts/test-atypical-padding.py:238]: This is probably just my limited TLS knowledge, but.. Why are we adding EMS extension to ServerHello in couple of places, but in most cases, we are not adding it?

**Review by ep69 — APPROVED** [review]
I left comments, but not strictly requiring changes.

File: scripts/test-atypical-padding.py (hunk 2 above)
Comment (tomato42) [reply scripts/test-atypical-padding.py:238]: if we specify the server extension in ExpectServerHello,we need to specify all the ones that server needs to send, if we don't specify any, then server can respond with any valid

File: scripts/test-atypical-padding.py (hunk 1 above)
Comment (tomato42) [reply scripts/test-atypical-padding.py:452]: I'll leave `ext` as-is, as that's what we use basically everywhere for client hello, but good point on `extensions`, changed it to `srv_ext`, as it is in other scripts

**Review by ep69 — APPROVED** [review]
LGTM

**Pull request merged**